ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -53,6 +54,9 @@
   private final AtomicLong endOffset;
   private final AtomicLong refCount = new AtomicLong(0);
   private final AtomicBoolean open = new AtomicBoolean(true);
+  static final int THREAD_LOCAL_BYTE_BUFFER_SIZE = 1024 * 1024;
+  private static ThreadLocal<ByteBuffer> threadLocalByteBuffer =
```

Comment:
this will create a buffer for every log segment - which is unnecessary because only the latest one is written to. We have to put some conditions around the creation (and cleanup of this buffer)